brayns/io/CircuitLoader.cpp

```diff
@@ -192,11 +200,19 @@ class CircuitLoader::Impl
                                                   source, metadata);
             modelDesc->setTransformation(transformation);
 
-            // unset the simulation handler once the model is removed
-            if (compartmentReport)
-                modelDesc->onRemoved([& scene = _parent._scene](const auto&) {
-                    scene.setSimulationHandler(nullptr);
+            if (simulationHandler)
+            {
+                auto& ap = _animationParameters;
+                auto handle = ap.addIsReadyCallback([simulationHandler] {
+                    return simulationHandler->isReady();
                 });
+                ap.setDt(simulationHandler->getDt());
+                ap.setUnit(simulationHandler->getUnit());
+                ap.setEnd(simulationHandler->getNbFrames());
```

Comment:
Multiple simulation handlers will not work unless they have the same dt and nbframes. Some consistency needs to be ensured. Also this code may be called concurrently (even if nonsensical), addIsReadyCallback needs  to be thread safe.

Moving this logic out of the constructor may also help.

Or prevent adding a second handler to the animation parameters.